lib/libzfs/os/freebsd/libzfs_zmount.c

```diff
@@ -128,8 +128,8 @@ do_unmount(const char *mntpt, int flags)
 	return (0);
 }
 
-int
-zfs_mount_delegation_check(void)
+boolean_t
+zfs_could_mount(void)
```

Comment:
The changed name may read better but I don't find it as meaningful.

Reply:
Do you have a better one?

Reply:
"could" is so vague as to be nearly meaningless. It tells me nothing about what the function does. 

The intent of the function is to check a specific condition necessary but not sufficient to mount. The existing name conveys that we are checking whether the user is permitted to perform the mount. You can see this in the history too, it was originally named `zfs_user_can_mount` before the name changed recently with 4bc721965ff272ad59c6e8a130a544e1e2c01b34.
```diff
-               if (geteuid() != 0) {
+               if (zfs_mount_delegation_check()) {
```
The implementation itself still needs improvement. To illustrate this, consider the following:
```
xeon-freebsd ➜  ~ sudo zfs allow ryan create,mount storage/zoftests
xeon-freebsd ➜  ~ zfs create storage/zoftests/foo
cannot mount 'storage/zoftests/foo': Insufficient privileges
filesystem successfully created, but not mounted
xeon-freebsd ➜  ~
```
I get the "filesystem successfully created, but not mounted" error message, rather than the more appropriate "filesystem successfully created, but it may only be mounted by root" message. I am not allowed to mount as a user by the system configuration, which can change:
```
xeon-freebsd ➜  ~ sudo sysctl vfs.usermount=1
vfs.usermount: 0 -> 1
xeon-freebsd ➜  ~ zfs create storage/zoftests/bar
xeon-freebsd ➜  ~
```
So I'd suggest leaving the function alone, thus allowing the FreeBSD implementation to be improved. The existing name could be debated back and forth about the term "delegation" versus something that doesn't overlap with other ZFS terminology, but it at least conveys we're checking if policy allows the user to mount.